### PR TITLE
Adding system diagram and brief explanation of workings

### LINKED ISSUE #206: Add a system diagram
Add a system architecture diagram in the docs that explains the inner workings of torchserve and its components - the front end clients, job processing queue, backend workers. Details should also cover the details about the interplay between the number_of_netty_threads, netty_client_threads, workers/model, job_queue_size.

### PR DESCRIPTION
**Pull request by maaquib**
## Description

Adding system diagram and brief explanation of workings

Closes #206 

## Type of change

Please delete options that are not relevant.

- [X] This change requires a documentation update

## Checklist:

- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by dhanainme** [comment]
@chauhang can you please confirm if this looks good. I can merge this for v0.0.1 release upon confirmation.

**Comment by maaquib** [comment]
> LGTM.
> 
> One point is I'd change the subtopic to `Concurrency And Number of Workers` (basically something that indicates this is about configuring concurrency) ?? @chauhang thoughts?

Done!!

File: docs/configuration.md [hunk 1]
```diff
@@ -188,8 +188,8 @@ By default, TorchServe uses all available GPUs for inference. Use `number_of_gpu
 Most of the following properties are designed for performance tuning. Adjusting these numbers will impact scalability and throughput.
 
 * `enable_envvars_config`: Enable configuring TorchServe through environment variables. When this option is set to "true", all the static configurations of TorchServe can come through environment variables as well. Default: false
-* `number_of_netty_threads`: number frontend netty thread. Default: number of logical processors available to the JVM.
-* `netty_client_threads`: number of backend netty thread. Default: number of logical processors available to the JVM.
+* `number_of_netty_threads`: number frontend netty thread. This specifies the numer of threads in the child [EventLoopGroup](https://livebook.manning.com/book/netty-in-action/chapter-8) of the frontend netty server. This group provides EventLoops for processing Netty Channel events (namely inference and management requests) from accepted connections. Default: number of logical processors available to the JVM.
```
Comment (mycpuorg) [inline docs/configuration.md:191]: Maybe add a bit of background about how Netty is used in TorchServe

File: README.md [hunk 2]
```diff
@@ -4,6 +4,16 @@ TorchServe is a flexible and easy to use tool for serving PyTorch models.
 
 **For full documentation, see [Model Server for PyTorch Documentation](docs/README.md).**
 
+## TorchServe Architecture
+![Architecture Diagram](https://user-images.githubusercontent.com/880376/83180095-c44cc600-a0d7-11ea-97c1-23abb4cdbe4d.jpg)
+
+### Terminology:
+* **Frontend**: The request/response handling component of TorchServe. This portion of the serving component handles both request/response coming from clients as well manages the models lifecycle.
+* **Model Workers**: These workers are responsible for running the actual inference on the models. These are actual running instances of the models.
```
Comment (dhaniram-kshirsagar) [inline README.md:12]: In the diagram, we should name Thread1/Thread2 as Worker1/Worker2.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,16 @@ TorchServe is a flexible and easy to use tool for serving PyTorch models.
...
+* **Model**: Models could be a `script_module` (JIT saved models) or `eager_mode_models`. These models can provide custom pre- and post-processing of data along with any other model artifacts such as state_dicts. Models can be loaded from cloud storage or from local hosts.
+* **Plugins**: These are custom endpoints or authz/authn or batching algorithms that can be dropped into TorchServe at startup time.
+* **Model Store**: This is a directory in which all the loadable models exist.
+
```
Comment (dhaniram-kshirsagar) [inline README.md:16]: Though not related -> 'Install Java 11' step has been repeated in 'install using pip' and 'install using conda'.

File: docs/configuration.md (hunk 1 above)
Comment (chauhang) [reply docs/configuration.md:191]: Yes that will be helpful to get some clarity on the internal workings and how to think about tuning the different threads related configuration parameters

**Pull request merged**